### PR TITLE
Pipeline howto 0.3.0

### PR DESCRIPTION
**Pull request by stoader**

### CODE REVIEW

File: docs/pipeline-howto.md [hunk 1]
```diff
@@ -1,261 +1,128 @@
...
 
-* Finish the wizard to create a `Control Plane` instance.
-* Take note of the PublicIP of the created Stack. We refer to this as the PublicIP of `Control Plane`.
+Setup your Pipeline GitHub OAuth application according to [this guilde](./github-app.md)
 
-    <a href="images/howto/CloudFormulationDetails5.png"><img src="images/howto/CloudFormulationDetails5.png" height="250"></a>
+### Deploy Control Plane
 
-* Go back to the earlier created GitHub OAuth application and modify it. Set the `Authorization callback URL` field to `http://{control_plane_public_ip}/auth/github/callback`
+Deploy `Control Plane` using [Pipeline Control Plane Launcher](https://github.com/banzaicloud/pipeline-cp-launcher/tree/0.3.0) to the one of the supported cloud providers where you would like to run your CI/CD flow.
```
Comment (baluchicken) [inline docs/pipeline-howto.md]: I think the "the" is not required here after the "to the one of the"

File: docs/pipeline-howto.md [hunk 2]
```diff
@@ -1,261 +1,128 @@
...
-     * PipelineImageTag - specify `0.2.0` for using current stable Pipeline release.
-
-     <a href="images/howto/AdvencedPipOpt.png"><img src="images/howto/AdvencedPipOpt.png" height="150"></a>
-
-  * **Slack Credentials**
-       * this section is optional. Complete this section to receive  cluster related alerts through a [Slack](https://slack.com) push notification channel.
-
-  * **Alert SMTP Credentials**
-     * this section is optional. Fill this section to receive cluster related alerts through email.
+Both Pipeline and CICD flow requires GitHub OAuth authentication and for this a OAuth application must be setup on GitHub.
```
Comment (baluchicken) [inline docs/pipeline-howto.md]: an OAuth app

File: docs/pipeline-howto.md [hunk 3]
```diff
@@ -1,261 +1,128 @@
...
+Navigate to `http://{control_plane_public_ip}/auth/github/login` in your web browser and grant access for the organizations that contain the GitHub repositories that you want to hook into the CI/CD workflow. Then click authorize access.
+
+If the login and authorization succeeds the user is redirected to the CI/CD UI.
 
 All the services of the Pipeline may take some time to fully initialize, thus the page may not load at first. Please give it some time and retry.
 
 ### Hook repositories to CI/CD flow
 
-Navigate to `http://{control_plane_public_ip}`  - this will bring you to the CI/CD user interface. Select `Repositories` from top left menu. This will list all the repositories that the Pipeline has access to.
+Navigate to `http://{control_plane_public_ip}/`  - this will bring you to the CI/CD user interface. Select `Repositories` from top left menu. This will list all the repositories that the Pipeline has access to.
```
Comment (baluchicken) [inline docs/pipeline-howto.md]: Sometimes, it is not enough for me to select the repositories I also need to hit the synchronise button too.

Comment (bonifaido) [reply docs/pipeline-howto.md]: This one has beed fixed just yesterday in https://github.com/banzaicloud/pipeline/pull/276

File: docs/pipeline-howto.md [hunk 4]
```diff
@@ -1,266 +1,134 @@
...
+The following sections may need to be customized:
 
-For setting up your own spark application for the workflow you can start from the `.pipeline.yml` configuration file from `spark-pi-example` and customize it.
+- the cluster where your application executed
 
-The following sections needs to be modified:
+  ```yaml
+  create_cluster:
+    ...
+    cluster_name: "[[your-cluster-name]]"
```
Comment (csendesm) [inline docs/pipeline-howto.md]: The cluster name must be a match of regex '(?:[a-z](?:[-a-z0-9]{0,38}[a-z0-9])?)' (only alphanumerics and '-' allowed, must start with a letter and end with an alphanumeric, and must be no longer than 40 characters)

Comment (stoader) [reply docs/pipeline-howto.md]: @csendesm I added this to the doc

**Pull request merged**